third_party/amd/lib/TritonAMDGPUTransforms/RescheduleOps.cpp

```diff
@@ -66,11 +79,60 @@ struct Node {
 
 private:
   Operation *op;
+  int32_t priority;
   llvm::SetVector<Node *> realChildren;
   llvm::SetVector<Node *> artificialChildren;
   llvm::SetVector<Node *> parents;
 };
 
+struct PriorityStrategy {
+  virtual void set(llvm::SmallVector<std::unique_ptr<Node>> &nodes) = 0;
+};
+
+struct BasicDotOpsPriorityStrategy : public PriorityStrategy {
+  void set(llvm::SmallVector<std::unique_ptr<Node>> &nodes) override {
+    SmallVector<Node *> dots;
+    SmallVector<Node *> loads;
+    for (auto &node : nodes) {
+      if (dyn_cast<triton::DotOp>(node->getOp()))
+        dots.push_back(node.get());
+      if (dyn_cast<triton::LoadOp>(node->getOp()))
+        loads.push_back(node.get());
+    }
+
+    // Make sure that prio is never equal to `0` for dotOps
+    int32_t dotPrio = dots.size() + 10;
```

Comment:
10 seems to be a "magic number". Please name and explain.

Reply:
done